File: 3189-find-champion-ii/3189-find-champion-ii.py
class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        inDeg = [0] * n
        for i in range(len(edges)):
            v = edges[i][1]
            inDeg[v] += 1
        
        cnt, ans = 0, -1
        for i in range(len(inDeg)):
            if inDeg[i] == 0:
                cnt += 1
                ans = i
        return ans if cnt == 1 else -1

File: 3189-find-champion-ii/3189-find-champion-ii.java
class Solution {
    public int findChampion(int n, int[][] edges) {
        int[] inDeg = new int[n];
        for (int i = 0; i < edges.length; ++i) {
            int v = edges[i][1];
            inDeg[v]++;
        }

        int cnt = 0, ans = -1;
        for (int i = 0; i < inDeg.length; ++i) {
            if (inDeg[i] == 0) {
                ++cnt;
                ans = i;
            }
        }
        return cnt == 1 ? ans : -1;
    }
}